packages/StaticTypeMapper/PhpParser/FullyQualifiedNodeMapper.php

```diff
@@ -39,6 +57,35 @@ public function mapToPHPStan(Node $node): Type
         return new FullyQualifiedObjectType($fullyQualifiedName);
     }
 
+    private function resolvePossibleAliasedObjectType(FullyQualified $fullyQualified): ?AliasedObjectType
+    {
+        $file = $this->currentFileProvider->getFile();
+        if (! $file instanceof File) {
+            return null;
+        }
+
+        $oldTokens = $file->getOldTokens();
+        $startTokenPos = $fullyQualified->getStartTokenPos();
+
+        if (! isset($oldTokens[$startTokenPos][1])) {
+            return null;
+        }
+
+        $type = $oldTokens[$startTokenPos][1];
+        if (str_contains($type, '\\')) {
+            return null;
+        }
+
```

Comment:
I tried that, when the alias is not used, it will always fail to compare as it always got `FullyQualified` type for `Param` type, the issue seems similar like in PR https://github.com/rectorphp/rector-src/pull/1093

Do you have an idea how to resolve it?

Reply:
The use cases are:

- Aliased exists and used as Param type --> follow aliased name
- Aliased exists but not used as Param type --> follow original name

without token check, it can't resolve the Param name, even tried with `ORIGINAL_NAME`, `RESOLVED_NAME`, all pointed to FullyQualified name.

This issue not happen in rector 0.11.60, only in the dev-main, probably because of use PHPStan 1.0

Reply:
@TomasVotruba another possible solution is possibly use `NodeTypeResolver->getType()`, it seems require refactoring on `ObjectTypeSpecifier->matchAliasedObjectType()`:

https://github.com/rectorphp/rector-src/blob/f6afff9f24bb6ac3462699cc466445c832a18fef/rules/TypeDeclaration/PHPStan/Type/ObjectTypeSpecifier.php#L82

Reply:
@TomasVotruba using `NodeTypeResolver->getType()` seems won't work as well, as I tried, it will always back to `AliasedObjectType`, even the alias is never used, that will fail the expectation of the second fixture https://github.com/rectorphp/rector-src/pull/1125/commits/30c65ea358330827fd9f1e48de0bbbe8f794764c

While using this token check, it can working ok

Reply:
@TomasVotruba I found a solution for it 🎉 , on loop uses, when alias found, we check next uses if there will be same classname definition with the fully qualified name next use https://github.com/rectorphp/rector-src/pull/1125/commits/ca8811def28911be16817cf6a585e2790cb02f2a

Reply:
@TomasVotruba using `NodeTypeResolver->getType()` seems broke other tests https://github.com/rectorphp/rector-src/runs/4082898936?check_suite_focus=true#step:5:67

I am reverting to back to original token solution

Reply:
@TomasVotruba the change to use `NodeTypeResolver->getType()` reverted, back to token usage. This is the solution I can found for now.

Reply:
I see, thanks for exploring these option. I would assume `matchAliasedObjectType()` would work. Mabye it needs a fix sort of.

For now, let's merge it as it is. Thank you :+1: